### PR TITLE
Update dockercomposefiles with additional release notes changes in 2.12.0 release

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Update dockercomposefiles with additional release notes changes in 2.12.0 release

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/4115

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker/release/dockercomposefiles/docker-compose-2.x.yml [hunk 1]
```diff
@@ -2,40 +2,40 @@
...
     environment:
       - cluster.name=opensearch-cluster
       - node.name=opensearch-node1
       - discovery.seed_hosts=opensearch-node1,opensearch-node2
       - cluster.initial_cluster_manager_nodes=opensearch-node1,opensearch-node2
-      - bootstrap.memory_lock=true  # along with the memlock settings below, disables swapping
-      - OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m    # minimum and maximum Java heap size, recommend setting both to 50% of system RAM
+      - bootstrap.memory_lock=true # along with the memlock settings below, disables swapping
+      - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m" # minimum and maximum Java heap size, recommend setting both to 50% of system RAM
       - OPENSEARCH_INITIAL_ADMIN_PASSWORD=${OPENSEARCH_INITIAL_ADMIN_PASSWORD}    # Sets the demo admin user password when using demo configuration, required for OpenSearch 2.12 and higher
```
Comment (prudhvigodithi) [inline docker/release/dockercomposefiles/docker-compose-2.x.yml:14]: Do we need double quotes here? coming from https://github.com/opensearch-project/opensearch-build/issues/4115#issuecomment-1953322664.

Comment (peterzhuamazon) [reply docker/release/dockercomposefiles/docker-compose-2.x.yml:14]: This is to keep consistent with the changes @AMoo-Miki and I doing on the website version.
The website version has been using double quotes the whole time.
Just to be consistent right here.

Comment (prudhvigodithi) [reply docker/release/dockercomposefiles/docker-compose-2.x.yml:14]: Thanks, just tested `OPENSEARCH_INITIAL_ADMIN_PASSWORD=${OPENSEARCH_INITIAL_ADMIN_PASSWORD} ` works on my local.

**Review by prudhvigodithi — APPROVED** [review]

**Review by nateynateynate — APPROVED** [review]
Looks good to me.

File: docker/release/dockercomposefiles/docker-compose-2.x.yml (hunk 1 above)
Comment (peterzhuamazon) [reply docker/release/dockercomposefiles/docker-compose-2.x.yml:14]: Tweaked to use the original setup as the download page one have yaml linting issues.

**Pull request merged**